### ISSUE TITLE
Bug: Kuzu CLI hangs with `iri` property

### ISSUE THREAD

**Issue by sapalli2989** [kuzudb/kuzu#3652]
### Kùzu version

v0.4.2

### What operating system are you using?

Fedora 39

### What happened?

```cypher
CREATE NODE TABLE V (id INT64, PRIMARY KEY(id));
CREATE REL TABLE E (FROM V TO V, iri STRING);
MATCH (v:V)-[e:E]->(v2:V) RETURN *;
```
Last statement will hang. Replacing `iri` by `p` works as usual.

**Comment by andyfengHKU** [comment]
Hi @sapalli2989, the bug is because we reserve `iri` as a built-in RDF property. And creating property under name `iri` will trigger logic specific to RDF cases.

I have fixed this in #3668 and we now throw exceptions if `iri` is used as a property name.

**Comment by sapalli2989** [comment]
Yeah thought so, that `iri` is a special case. Hence sounds like a good solution to get an according error message.

Not sure, how often are tables with `iri` property outside RDFGraphs, but out of curiosity: Can't you resort to some kind of table meta data to decide, if table at hand belongs to RDFGraphs or not? This would get rid of another conditional property check.
